### PR TITLE
Use trimpath for compatibility with go-ipfs v0.7.0

### PR DESCRIPTION
**Pull request by aschmahmann**
This PR changes the build flags used when building plugins.

There are 3 types of go-ipfs we support building plugins against:
1) The release binaries distributed at dist.ipfs.io
2) A locally built go-ipfs
3) A Git referred version (e.g. commit/branch/tag) of go-ipfs

Previously:
Use the `-asmflags=all=-trimpath="$(GOPATH)" -gcflags=all=-trimpath="$(GOPATH)"` flags no matter which version of the plugin we're building against.

This was a pain when building against a Git commit/branch/tagged version of go-ipfs since the complex set of flags above had to be passed into `go get github.com/ipfs/go-ipfs/cmd/ipfs@{commit|branch|tag}`. Additionally, the use of GOPATH here may not be correct depending on how the users' directory structure is set up.

Now:
Use the `-asmflags=all=-trimpath="$(GOPATH)" -gcflags=all=-trimpath="$(GOPATH)"` **only** when building against a locally build copy of go-ipfs, otherwise use the `-trimpath` flag instead.

This means we can run `go get -trimpath github.com/ipfs/go-ipfs/cmd/ipfs@{commit|branch|tag}` and just build the plugin with `-trimpath` (as done in the Makefile here) and things will work.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.12
 
 require (
 	github.com/ipfs/go-datastore v0.4.4
-	github.com/ipfs/go-ipfs v0.5.1
+	github.com/ipfs/go-ipfs v0.6.0
```
Comment (Stebalien) [inline go.mod]: 0.7?

Comment (Stebalien) [reply go.mod]: Ah, I see.

File: go.mod [hunk 2]
```diff
@@ -3,8 +3,8 @@ module github.com/ipfs/go-ipfs-example-plugin
 go 1.12
 
 require (
-	github.com/ipfs/go-datastore v0.4.4
-	github.com/ipfs/go-ipfs v0.5.1
+	github.com/ipfs/go-datastore v0.4.5
+	github.com/ipfs/go-ipfs v0.7.0-rc2
```
Comment (aschmahmann) [inline go.mod]: Once v0.7.0 is released will update this.

**Comment by aschmahmann** [comment]
@Stebalien updated, take a look when you get a chance :pray:

**Review by Stebalien — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (aschmahmann) [reply go.mod]: Done

**Pull request merged**